aten/src/ATen/Device.h

```diff
@@ -60,10 +46,8 @@ struct Device {
   /// `<device-index>` optionally specifies a device index.
   /* implicit */ Device(const std::string& device_string);
 
-  /// Constructs a new `Device` from a `Backend` (which is converted to a
-  /// `DeviceType`, if possible) and an optional device index.
-  /* implicit */ Device(Backend backend, int32_t index = -1)
-      : Device(backend_to_type(backend), index) {}
+  /* implicit */ Device(BackendOrDeviceType backend_or_device_type)
+    : Device(backend_or_device_type.operator DeviceType()) {}
```

Comment:
`backend_or_device_type.operator DeviceType()` 😮

Reply:
I guess a `static_cast` would have been sufficient here. In another version of the patch, calling the operator was mandatory, because `static_cast` would also consider the constructors (which caused ambiguity).